etcdserver/server.go

```diff
@@ -204,6 +204,13 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 		cl *cluster
 	)
 
+	if mkerr := fileutil.MkdirAllIfNotExist(cfg.DataDir); mkerr != nil {
+		return nil, fmt.Errorf("cannot create data directory: %v", mkerr)
+	}
+	if wderr := fileutil.IsDirWriteable(cfg.DataDir); wderr != nil {
```

Comment:
It's like the directory equivalent of `touch` (ignoring atime/mtime stuff), isn't it? Create if not present, otherwise return true if writeable.